tests/dummy/app/controllers/components-examples/flexberry-objectlistview/selected-rows.js

```diff
@@ -21,7 +30,27 @@ export default ListFormController.extend({
 
   actions: {
     configurateSelectedRows(selectedRecords) {
-      this.set('countSelectedRows', selectedRecords.length);
+      if (this.get('allSelect')) {
+        this.set('countSelectedRows', this.model.meta.count);
+      } else {
+        this.set('countSelectedRows', selectedRecords.length);
+      }
     }
-  }
+  },
+
+  _selectAll(componentName, selectAllParameter) {
+    if (componentName === this.get('componentName')) {
+      this.set('allSelect', selectAllParameter);
+    }
+  },
+
+  init: function() {
+    this.get('objectlistviewEventsService').on('updateSelectAll', this, this._selectAll);
+    this._super(...arguments);
+  },
+
+  willDestroyElement() {
```

Comment:
In this case, you need a [`willDestroy`](https://emberjs.com/api/ember/2.4/classes/Ember.Controller/methods/willDestroy?anchor=willDestroy) method.